### PR TITLE
OP-256 add license plugin and add headers

### PR DESCRIPTION
**Pull request by dbmalkovsky**
Per discussion in core's [PR 165](https://github.com/informatici/openhospital-core/pull/165) and opened up [OP-265](https://openhospital.atlassian.net/jira/software/c/projects/OP/issues/OP-256) to define an the license header comment.

The license plugin uses the same exclusion list as the one in the Core repository.

### CODE REVIEW

**Comment by mwithi** [comment]
Re-thinking, maybe we can hold this untile https://github.com/informatici/openhospital-api/tree/OP-6-implement-the-new-rest-api will be merged, then update this PR, what do you think? We are almost there anyway...

**Comment by dbmalkovsky** [comment]
That is fine with me; I'll close it and wait for you to give me the go ahead of that is OK with you.

**Comment by mwithi** [comment]
There's no need to close it, i've just labelled it

**Comment by dbmalkovsky** [comment]
Good idea; I'll just have to rerun the plugin and commit the new files once OP-6 is merged.

**Comment by mwithi** [comment]
> Good idea; I'll just have to rerun the plugin and commit the new files once [OP-6](https://openhospital.atlassian.net/browse/OP-6) is merged.

Yes, and after merging the pulled `develop` into `dbmalkovsky:OP-256-add-license-header`

**Comment by mwithi** [comment]
@dbmalkovsky I think we can proceed now with this (rebasing or make from scratch)

**Comment by dbmalkovsky** [comment]
@mwithi So the decision is **not** to wait for [OP-6](https://github.com/informatici/openhospital-api/tree/OP-6-implement-the-new-rest-api) to be merged?

**Comment by mwithi** [comment]
> @mwithi So the decision is **not** to wait for [OP-6](https://github.com/informatici/openhospital-api/tree/OP-6-implement-the-new-rest-api) to be merged?

OP-6 has been merged few hours ago #68, that's why I'm saying ;)

**Comment by dbmalkovsky** [comment]
Ah I didn't see that; sorry.

**Comment by dbmalkovsky** [comment]
@mwithi I have rebased the branch on develop and have added the license headers and all the tests passed.
Do you want me to reformat the files at the same time?

**Comment by dbmalkovsky** [comment]
I think this is it finally; rebased; reformat the files to match coding standards, and updated import statements to match standards too.

File: pom.xml [hunk 1]
```diff
@@ -109,21 +110,20 @@
...
-        </dependency> -->
+            <version>2.2.4</version>
+        </dependency>
         <dependency>
-            <groupId>org.glassfish</groupId>
+            <groupId>org.glassfish.web</groupId>
             <artifactId>javax.el</artifactId>
-            <!-- version>2.2.4</version> -->
-            <version>3.0.1-b08</version>
+            <version>2.2.4</version>
```
Comment (mwithi) [inline pom.xml:121]: Why it's downgrading again javax.el?

File: pom.xml [hunk 2]
```diff
@@ -109,21 +110,20 @@
...
         </dependency>
         <dependency>
             <groupId>org.projectlombok</groupId>
             <artifactId>lombok</artifactId>
-            <version>1.18.4</version>
+            <scope>provided</scope>
```
Comment (mwithi) [inline pom.xml:126]: Lombok version seems needed (in my case it won't build) but also it allows dependabot to provide suggestions.

**Review by mwithi — COMMENTED** [review]
Some questions on `pom.xml`...

Rest of the changes lgtm

File: pom.xml (hunk 1 above)
Comment (dbmalkovsky) [reply pom.xml:121]: Looks like javax.el should not be there and the glassfish dependency was a bad merge on my part.  The develop branch currently has:

```
        <!-- <dependency>
            <groupId>javax.el</groupId>
            <artifactId>javax.el-api</artifactId>
            <version>3.0.0</version>
        </dependency> -->
        <dependency>
            <groupId>org.glassfish</groupId>
            <artifactId>javax.el</artifactId>
            <!-- version>2.2.4</version> -->
            <version>3.0.1-b08</version>
        </dependency>
```
I'll fix that; good catch.

File: pom.xml (hunk 2 above)
Comment (dbmalkovsky) [reply pom.xml:126]: OK; I'll fix that one too.

**Pull request merged**